File: easy/happy-number/happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function (n) {
  const set = new Set();
  while (n !== 1) {
    n = getSumSquares(n);
    if (set.has(n)) {
      return false;
    }
    set.add(n);
  }
  return true;
};

var getSumSquares = function (n) {
  let result = 0;
  while (n > 0) {
    result += (n % 10) ** 2;
    n = Math.floor(n / 10);
  }
  return result;
};

File: easy/happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        viewed = set()
        while n != 1:
            n = self.getSumSquares(n)
            if n in viewed:
                return False
            viewed.add(n)
        return True

    def getSumSquares(self, n: int) -> int:
        result = 0
        while n > 0:
            result += (n % 10) ** 2
            n = n // 10
        return result
